content/en/docs/tasks/configure-pod-container/configure-gmsa.md

```diff
@@ -0,0 +1,302 @@
+{{< /note >}}
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Setup and configuration for GMSA
+Configuring GMSA credential specs in the cluster and configuring individual pods and containers to be able to use them requires several steps described in details below:
+1. Enable the `WindowsGMSA` feature gate on kubelet on Windows nodes.
+2. Install the GMSACredentialSpec CRD
+3. Create GMSA credential spec resources
+4. Install webhooks to expand and validate references to GMSA credential spec resources from pod specs
+5. Create cluster roles to allow service accounts to use specific GMSA credential spec resources
+6. Bind roles to specific service accounts to allow them to use GMSA credential spec resources
+7. Configure pods to use GMSA credential specs along with a service account authorized to use the GMSA credential specs
+
+### Enable the WindowsGMSA feature gate
+In the alpha state, the `WindowsGMSA` feature gate needs to be enabled on kubelet on Windows nodes. This is required to pass down the GMSA credential specs from the cluster scoped configurations to the container runtime. See [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) for an explanation of enabling feature gates. Please make sure `--feature-gates=WindowsGMSA=true` parameter exists in the kubelet.exe command line.
+
+### Install the GMSACredentialSpec CRD
+A [CustomResourceDefinition](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/) (CRD) for GMSA credential spec resources needs to be configured on the cluster to define the custom resource type `GMSACredentialSpec`. Download the GMSA CRD [YAML](https://github.com/kubernetes-sigs/windows-gmsa/blob/master/admission-webhook/deploy/gmsa-webhook.yml.tpl#L131-L148) and save it as gmsa-crd.yaml.
+Next, install the CRD with `kubectl apply -f gmsa-crd.yaml`
+
+### Create GMSA credspec resources
+With the GMSACredentialSpec CRD installed, GMSA credspec custom resources can now be configured. The GMSA credential spec does not contain secret or sensitive data. It is information that a container runtime can use to describe the desired GMSA of a container to Windows. The GMSA credspec resources can be generated in JSON format with a utility [PowerShell][] script. Following are the steps for generating a GMSA credspec YAML based on the JSON:
+1. Import the CredentialSpec module: `ipmo CredentialSpec.psm1`
+2. Create a credential spec in JSON format using `New-CredentialSpec`. To create a GMSA credspec named WebApp1: `New-CredentialSpec -Name WebApp1 -AccountName WebApp1 -Domain $(Get-ADDomain -Current LocalComputer)`
+3. Use `Get-CredentialSpec` to show the path of the JSON file. 
+4. Convert the credspec file from JSON to YAML format and apply the necessary header fields `apiVersion`, `kind`, `metadata` and `credspec` to make it a GMSACredentialSpec custom resource. An example based on a GMSA credspec named WebApp1 is below:
```

Comment:
We could package that in a nice powershell script, or even better add it to the `CredentialSpec.psm1` module. Happy to do it if you think that's a good idea?

Reply:
I agree it would be good to have that PS module give you the option of YAML output.

Reply:
Definitely a good idea @wk8 . Can you add it to https://github.com/kubernetes-sigs/windows-gmsa?